Makefile.clang

```diff
@@ -0,0 +1,238 @@
+CURRENT_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+CC := clang-16
+LD := ld.lld-16
+OBJCOPY := llvm-objcopy-16
+CFLAGS_LLVM := --target=riscv64 -march=rv64imac_zba_zbb_zbc_zbs -Wno-error=unused-command-line-argument
+CFLAGS := ${CFLAGS_LLVM} -fPIC -O3 -nostdinc -nostdlib -nostartfiles -fvisibility=hidden -I deps/ckb-c-stdlib -I deps/ckb-c-stdlib/libc -I deps -I deps/ckb-c-stdlib/molecule -I c -I build -I deps/secp256k1/src -I deps/secp256k1 -g -Wno-nonnull -Wno-unused-function
+LDFLAGS := -Wl,-static -fdata-sections -ffunction-sections -Wl,--gc-sections
```

Comment:
`-fdata-sections -ffunction-sections` are CFLAGS and only `-Wl,--gc-sections` is for linker.

Reply:
Here I copied it from: https://github.com/nervosnetwork/ckb-miscellaneous-scripts/blob/master/Makefile#L8

To be honest, I think this is not the only problem in the Original Makefile

Reply:
Fix it if possible. The binary can be larger than expected.